### PR TITLE
Add missing dependency jupyterhub

### PR DESCRIPTION
**Pull request by Ph0tonic**
Add the missing `jupyterhub` dependency required for the authentication.

### CODE REVIEW

**Review by volodymyrss — CHANGES_REQUESTED** [review]
The behavior without jupyterhub is still not defined, right? As we discussed, it would be good to have some reasonable exception in this case.

**Comment by Ph0tonic** [comment]
> The behavior without jupyterhub is still not defined, right? As we discussed, it would be good to have some reasonable exception in this case.
Done 👍, sorry for the weird commit history.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
 # Download service
 
-Backend server for [https://github.com/cta-epfl/ctadata](https://github.com/cta-epfl/ctadata).
\ No newline at end of file
+Backend server for [https://github.com/cta-epfl/ctadata](https://github.com/cta-epfl/ctadata).
+
+## Installation
+
+For `production` use to install dependencies just run :
```
Comment (volodymyrss) [inline README.md:7]: it's not certain that production use of this service will be always within jupyterhub.

```suggestion
For use as a JupyterHub service, run:
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+```
+poetry install
+```
+
+For `test` purpose without jupyterhub :
```
Comment (volodymyrss) [inline README.md:12]:
```suggestion
For standalone service without JupyterHub :
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+```
+poetry install --with dev --without jupyterhub
+```
```
Comment (volodymyrss) [inline README.md:15]:
```suggestion
```

Note, however, that in this mode the service will not check authorization and will by default fail to respond to any request for restricted data. However, if `CTADS_DISABLE_ALL_AUTH` application config is set, the service will authorize all requests: this way this mode can also be used for for testing. 
```

File: downloadservice/app.py [hunk 4]
```diff
@@ -82,17 +81,15 @@ def decorated(*args, **kwargs):
         if app.config['CTADS_DISABLE_ALL_AUTH']:
             return f("anonymous", *args, **kwargs)
         else:
+            if auth is None:
+                return "Error: Auth system not configured correctly", 500
```
Comment (volodymyrss) [inline downloadservice/app.py:85]:
```suggestion
                return "Unable to use to use jupyterhub to verify access to this service. At this time, the downloadservice  uses jupyterhub to control access to protected resources", 500
```

**Review by volodymyrss — CHANGES_REQUESTED** [review]
Couple of suggestions to account for the fact that the service might not be run by default as part of the jupyterhub. Though as now it relies on JH for configuring, it's not as likely. But in the future, we might need to adapt to make more meanigful no-JH mode.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,15 @@
...
+## Installation
+
+For use as a JupyterHub service, run:
+```
+poetry install
+```
+
+For standalone service without JupyterHub :
+```
+poetry install --with dev --without jupyterhub
```
Comment (volodymyrss) [inline README.md:14]: seems like I can not make a suggestion that includes ``` since it's used to limit suggestion 🤦. Could you add it?

Comment (Ph0tonic) [reply README.md:14]: Done 👍

**Review by volodymyrss — APPROVED** [review]

**Comment by volodymyrss** [comment]
Thanks!

**Pull request merged**